### DISCUSSION TITLE
PMI: best practice for handling undefined stereo centers?

### DISCUSSION THREAD

**Discussion by apahl** [rdkit/rdkit#5102] [General]
Dear all,

what would be considered best practice to calculate Principal Moment of Inertia (PMI) for structures that have undefined stereo centers?

I see 3 options, all of them do not seem ideal:

1. Exclude all structures with undefined stereo centers from the analysis.

2. Randomly choose one of the isomers at any given undefined stereo center (will give quite arbitrary results).

3. Enumerate all undefined stereo centers into all possible different structures and either
(a) calculate individual PMIs for the enumerated structures (would lead to an explosion of compound numbers) or 
(b) calculate an average PMI over all enumerated structures (where the different stereoisomers might cancel each other out in terms of their "3d-ness").

Currently I am tending towards 3a, and setting a limit on the number of undefined stereo centers (2-3).

Any recommendations would be welcome.

Best,
Axel

**Comment by greglandrum** [comment]
Hi @apahl, it's an interesting question, and I don't have a quick answer for it.

I do wonder if the range of differences in PMI caused by different stereochemistry is on average more or less significant than the differences caused by conformational variability.

That might make an interesting blog post... :-)

**Reply by bp-kelley** [reply]
I’ll bet a chunk of invertible nitrogens won’t make a difference, especially ones in rings with small substituents.

I’d be rather interested in the general ensemble variability as well.  I’d be happy to run the same on omega confs for a rotor driven ensemble.
